src/Microsoft.ML.Data/Data/RowCursorUtils.cs

```diff
@@ -407,9 +409,12 @@ public static ValueGetter<VBuffer<Single>> GetLabelGetter(SlotCursor cursor)
                 return cursor.GetGetter<Single>();
             if (type == NumberType.R8 || type is BoolType)
                 return GetVecGetterAs<Single>(NumberType.R4, cursor);
-            Contracts.Check(type.IsKey, "Only floating point number, boolean, and key type values can be used as label.");
+            if (!(type is KeyType keyType))
+            {
+                throw Contracts.Except("Only floating point number, boolean, and key type values can be used as label.");
+            }
```

Comment:
Minor comment: maybe we can do this in one line.

Reply:
Unfortunately we can't use `Contracts.Check` anymore because the C# compiler doesn't know that `Contracts.Check` will throw if the parameter evaluates to `false`. So it complains if you use `keyType` after the `Contracts.Check` line.